### PR TITLE
Feature/rendering shape

### PR DESCRIPTION
**Pull request by Nniy**

### CODE REVIEW

**Comment by Nniy** [comment]
@0xsuu Ｒｅｖｉｅｗ

File: src/ABEWindow.cpp [hunk 1]
```diff
@@ -25,6 +25,7 @@
 
 using std::invalid_argument;
 
+using abege::ABEShader;
```
Comment (0xsuu) [inline src/ABEWindow.cpp]: Remove this line since we don't use it.

File: src/ABEShader.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    glDeleteShader(vertex);
+    glDeleteShader(fragment);
+}
+
+void ABEShader::checkCompileErrors(GLuint shader, std::string type) {
+    GLint success;
+    GLchar infoLog[1024];
+    if (type == ABE_PROGRAM_COMPILE) {
+        glGetProgramiv(shader, GL_LINK_STATUS, &success);
+        if (!success) {
```
Comment (0xsuu) [inline src/ABEShader.cpp:73]: Stop tracking this issue!

File: resources/SimpleFragmentShader.fs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#version 330 core
+
+// Output data.
+out vec3 color;
+
+void main() {
+	// Output color = red.
+	color = vec3(1,0,0);
+}
```
Comment (0xsuu) [inline resources/SimpleFragmentShader.fs]: Add EOF

File: resources/SimpleVertexShader.vs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#version 330 core
+
+layout(location = 0) in vec3 vertexPosition_modelspace;
+
+void main() {
+    gl_Position.xyz = vertexPosition_modelspace;
+    gl_Position.w = 1.0;
+}
+
```
Comment (0xsuu) [inline resources/SimpleVertexShader.vs]: Remove redundant line.

File: include/ABEObject.h [hunk 5]
```diff
@@ -22,13 +22,35 @@
...
+
+#include <GL/glew.h>
+
 namespace abege {
 
 class ABEObject {
  public:
-    ABEObject(std::string name) : mName(name) {}
+    ABEObject(std::string name) : mName(name) {
+        // TODO: Remove these code.
```
Comment (0xsuu) [inline include/ABEObject.h]: Move this function into .cpp file.

File: src/ABEObject.cpp [hunk 6]
```diff
@@ -15,15 +15,31 @@
...
+#include "ABELogger.h"
 
 using std::make_pair;
 using std::pair;
 
 using abege::ABEObject;
 
 void ABEObject::render() {
-    // TODO(Wa): Add render function here.
+    glEnableVertexAttribArray(0);
```
Comment (0xsuu) [inline src/ABEObject.cpp]: Add ```glUseProgram(mShader->ID);``` here.

**Review by 0xsuu — CHANGES_REQUESTED** [review]

File: include/ABEObject.h (hunk 5 above)
Comment (0xsuu) [reply include/ABEObject.h]: done

File: src/ABEObject.cpp (hunk 6 above)
Comment (0xsuu) [reply src/ABEObject.cpp]: done

File: src/ABEWindow.cpp (hunk 1 above)
Comment (0xsuu) [reply src/ABEWindow.cpp]: done

File: resources/SimpleFragmentShader.fs (hunk 3 above)
Comment (0xsuu) [reply resources/SimpleFragmentShader.fs]: done

File: resources/SimpleVertexShader.vs (hunk 4 above)
Comment (0xsuu) [reply resources/SimpleVertexShader.vs]: done

**Review by 0xsuu — APPROVED** [review]

**Pull request merged**